### PR TITLE
Changelog for v2.187.0

### PR DESCRIPTION
**Pull request by vEpiphyte**
- changelog for v2.187.0
- update doctests to make sure changelog files are valid and do not contain multlline scalar values ;)
- update changelog tool to break multiple paragraphs of changelog text into discrete paragraphs when wrapping them.

```
desc: 
    I am a long line that is wrapped
    into multiple pars.
   
    I am another paragraphs.

    I am a another paragraph line that is wrapped
    into multiple partss.
```

### CODE REVIEW

File: scripts/doctests.py [hunk 1]
```diff
@@ -36,6 +50,24 @@ def check_rstorm(dirn):
...
+            continue
+        fp = os.path.abspath(os.path.join(dirn, fn))
+        print(f'Checking {fp}')
+        with open(fp, 'rb') as fd:
+            bytz = fd.read()
+        # Do we have multi-line scalers?
+        if re.findall(r'[a-z0-9]\:\s+\|\s*\n', bytz.decode('utf8'), flags=re.IGNORECASE):
+            raise ValueError(f'multiline scaler detected in {fp}')
+        # Just asserting we are valid yaml to start with.
+        data = yaml.load(bytz, yaml.CSafeLoader)
```
Comment (MichaelSquires) [inline scripts/doctests.py:66]: CSafeLoader isn't guaranteed to be present. Maybe `s_common.Loader` instead (which does the fallback dynamically)?

Comment (vEpiphyte) [reply scripts/doctests.py]: Just switched to SafeLoader.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -6,6 +6,42 @@
...
+
+Automatic Migrations
+--------------------
+- WARNING - It is strongly advised to perform a backup before upgrading to or
+  above this version. The ``it:sec:cpe`` migration described below WILL remove
+  invalid ``it:sec:cpe`` and some associated nodes from the Cortex.
+
+  Migrate invalid ``it:sec:cpe`` nodes if possible. Migration of these nodes
+  will only be successful if one of the CPE 2.3 (primary property) or the CPE
+  2.2 (``:v2_2``) strings are valid CPEs.  If both CPE strings are invalid, the
```
Comment (Cisphyx) [inline CHANGELOG.rst:20]:
```suggestion
  2.2 (``:v2_2``) strings are valid CPEs. If both CPE strings are invalid, the
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -6,6 +6,42 @@
...
+  node will be removed from the cortex and stored in a Cortex queue
```
Comment (Cisphyx) [inline CHANGELOG.rst:21]:
```suggestion
  node will be removed from the Cortex and stored in a Cortex queue
```

**Review by Cisphyx — APPROVED** [review]

File: CHANGELOG.rst [hunk 4]
```diff
@@ -6,6 +6,42 @@
...
+  the same mistake consistently).
+
+  During migration or removal of invalid ``it:sec:cpe`` nodes, referencing
+  nodes with readonly properties will be removed and also stored in the queue.
+  We are unable to automatically migrate these nodes due to the dynamic nature
+  of their construction.
+- See :ref:`datamigration` for more information about automatic migrations.
+
+Features and Enhancements
+-------------------------
```
Comment (MichaelSquires) [inline CHANGELOG.rst:36]:
```suggestion
-------------------------
- Update the parsing of CPE 2.2 and CPE 2.3 strings to be strict according
  to the CPE specification (NISTIR 7695).
```

**Review by MichaelSquires — DISMISSED** [review]

**Review by MichaelSquires — APPROVED** [review]

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**